pkg/cmd/dump.go

```diff
@@ -0,0 +1,200 @@
+	}
+	fmt.Fprintf(out, "Found %d platforms:\n", len(pls.Items))
+	for _, p := range pls.Items {
+		ref := p
+		pdata, err := kubernetes.ToYAML(&ref)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(out, "---\n%s\n---\n", string(pdata))
+	}
+
+	its, err := camelClient.CamelV1().Integrations(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Found %d integrations:\n", len(its.Items))
+	for _, integration := range its.Items {
+		ref := integration
+		pdata, err := kubernetes.ToYAML(&ref)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(out, "---\n%s\n---\n", string(pdata))
+	}
+
+	iks, err := camelClient.CamelV1().IntegrationKits(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Found %d integration kits:\n", len(iks.Items))
+	for _, ik := range iks.Items {
+		ref := ik
+		pdata, err := kubernetes.ToYAML(&ref)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(out, "---\n%s\n---\n", string(pdata))
+	}
+
+	cms, err := c.CoreV1().ConfigMaps(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Found %d ConfigMaps:\n", len(cms.Items))
+	for _, cm := range cms.Items {
+		ref := cm
+		pdata, err := kubernetes.ToYAML(&ref)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(out, "---\n%s\n---\n", string(pdata))
+	}
+
+	deployments, err := c.AppsV1().Deployments(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Found %d deployments:\n", len(iks.Items))
+	for _, deployment := range deployments.Items {
+		ref := deployment
+		data, err := kubernetes.ToYAML(&ref)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(out, "---\n%s\n---\n", string(data))
+	}
+
+	lst, err := c.CoreV1().Pods(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(out, "\nFound %d pods:\n", len(lst.Items))
+	for _, pod := range lst.Items {
+		fmt.Fprintf(out, "name=%s\n", pod.Name)
+		dumpConditions("  ", pod.Status.Conditions, out)
+		fmt.Fprintf(out, "  logs:\n")
+		var allContainers []v1.Container
+		allContainers = append(allContainers, pod.Spec.InitContainers...)
+		allContainers = append(allContainers, pod.Spec.Containers...)
+		for _, container := range allContainers {
+			pad := "    "
+			fmt.Fprintf(out, "%s%s\n", pad, container.Name)
+			err := dumpLogs(ctx, c, fmt.Sprintf("%s> ", pad), ns, pod.Name, container.Name, out)
+			if err != nil {
+				fmt.Fprintf(out, "%sERROR while reading the logs: %v\n", pad, err)
+			}
+		}
+	}
+	return nil
+}
+
+func dumpConditions(prefix string, conditions []v1.PodCondition, out *os.File) {
+	for _, cond := range conditions {
+		fmt.Fprintf(out, "%scondition type=%s, status=%s, reason=%s, message=%q\n", prefix, cond.Type, cond.Status, cond.Reason, cond.Message)
+	}
+}
+
+func dumpLogs(ctx context.Context, c client.Client, prefix string, ns string, name string, container string, out *os.File) error {
+	lines := int64(50)
```

Comment:
Maybe adding a command option could be useful. Sometimes errors appears after the 50 lines, and it's convenient for users to increase the amount of log lines retrieved.

Reply:
Will do.

Reply:
Done